k8s/k8s.go

```diff
@@ -1084,6 +1121,65 @@ func (k *k8sOps) UpdateDaemonSet(ds *apps_api.DaemonSet) error {
 
 // DaemonSet APIs - END
 
+// Job APIs - BEGIN
+func (k *k8sOps) CreateJob(job *batch_v1.Job) (*batch_v1.Job, error) {
+	if err := k.initK8sClient(); err != nil {
+		return nil, err
+	}
+
+	return k.client.Batch().Jobs(job.Namespace).Create(job)
+}
+
+func (k *k8sOps) GetJob(namespace, name string) (*batch_v1.Job, error) {
+	if err := k.initK8sClient(); err != nil {
+		return nil, err
+	}
+
+	return k.client.Batch().Jobs(namespace).Get(name, meta_v1.GetOptions{})
+}
+
+func (k *k8sOps) DeleteJob(namespace, name string) error {
+	if err := k.initK8sClient(); err != nil {
+		return err
+	}
+
+	policy := meta_v1.DeletePropagationForeground
+	return k.client.Batch().Jobs(namespace).Delete(name, &meta_v1.DeleteOptions{
+		PropagationPolicy: &policy,
+	})
+}
+
+func (k *k8sOps) ValidateJob(namespace, name string, timeout time.Duration) error {
+	t := func() (interface{}, bool, error) {
+		job, err := k.GetJob(namespace, name)
+		if err != nil {
+			return nil, true, err
+		}
+
+		if job.Status.Failed > 0 {
+			return nil, false, fmt.Errorf("job: [%s] %s has %d failed pod(s)", namespace, name, job.Status.Failed)
+		}
+
+		if job.Status.Active > 0 {
+			return nil, true, fmt.Errorf("job: [%s] %s still has %d active pod(s)", namespace, name, job.Status.Active)
+		}
+
+		if job.Status.Succeeded == 0 {
```

Comment:
will it happen that there are not failed or active pods and no successful pods? If yes, should we return true even if there is just 1 successful? Should we not validate that all succeeded?

Reply:
That case wouldn't occur. If it did, it would be a bug on the k8s api side since it returned an inconsistent job object.

For e.g consider a `Job`  has 3 pods doing it's work. If the first one completed successfully, `job.Status.Succeeded` will be 1. At that moment, `job.Status.Active` should be 2.